metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_v2.py

```diff
@@ -512,26 +525,31 @@ def _process_database(
         if self.config.include_technical_schema:
             yield from self.gen_database_containers(snowflake_db)
 
-        # Use database and extract metadata from its information_schema
-        # If this query fails, it means, user does not have usage access on database
+        # Extract metadata from its information_schema
         try:
-            self.query(conn, SnowflakeQuery.use_database(db_name))
             snowflake_db.schemas = self.data_dictionary.get_schemas_for_database(
                 conn, db_name
             )
+        except SnowflakePermissionError as e:
+            error_msg = f"Failed to list schemas for database {db_name}. Please check permissions."
+            # Ideal implementation would use PEP 678 – Enriching Exceptions with Notes
+            raise SnowflakePermissionError(error_msg) from e.__cause__
```

Comment:
is the intended behavior to throw an error that isn't caught in this method?

my impression was that the outer `except Exception as e:` was sufficient, and we should use `isinstance(e, SnowflakePermissionError)` for the special casing

Reply:
done

Reply:
Moved the handling of `SnowflakePermissionError` inside `except Exception`.

The intended behaviour is to bubble up these permission errors and making them fatal.